src/api/controllers/groupController.ts

```diff
@@ -0,0 +1,53 @@
+import { BaseController } from './baseController.js';
+import { RequestWithGroup } from '../../types/requests.js';
+import { NextFunction, Request, Response } from 'express';
+import { GroupService } from '../../services/groupService.js';
+import { BaseGroup } from '../../types/group.js';
+import { EntityNotFoundError } from '../../core/errors/entityNotFoundError.js';
+
+
+export class GroupController extends BaseController {
+    private groupService: GroupService;
+
+    constructor(groupServiceInjected: GroupService) {
+        super();
+        this.groupService = groupServiceInjected;
+    }
+
+    async createHandler(req: RequestWithGroup, res: Response, next: NextFunction) {
+        const group: BaseGroup = req.body;
+        try {
+            const createdGroup = await this.groupService.create(group);
+            this.success(res, createdGroup);
+        } catch (e) {
+            next(e);
+            return;
+        }
+    }
+    async getAllGroups(req: Request, res: Response, next: NextFunction) {
+        try {
+            const allGroups = await this.groupService.getAll();
+            if (!allGroups) {
```

Comment:
I think it is useless condition. Let's imaging the case, you want to check the list of products in some e-commerce stock, open the page and receive an error... "Product can not be found.".

I understand that's totally separated story but managing the response structure it is also very importing and particularly for this case, I would recommended to have something like:

```
this.success(res, {
     success: true,
     data: {
         groups: allGroups
     }
});
``` 

and in the feature, you will probably need to add the pagination and filtration by keywords for you output, so, the above structure will suite that case with only simple change, just adding to `data` a few properties, like:

```
const { limit=10, offset=0, keywords } = req.query
const groups = await this.groupService.getAll({ limit, offset, keywords });
...
this.success(res, {
     success: true,
     data: {
         limit,
         offset,
         keywords,
         groups: allGroups,
      }
}
```